src/addon-manager.js

```diff
@@ -760,11 +761,13 @@ class AddonManager extends EventEmitter {
       this.adapters.delete(a.id);
     }
 
+    // Give the process 5 seconds to exit before killing it.
     const cleanup = () => {
-      const plugin = this.getPlugin(packageName);
-      if (plugin) {
-        plugin.kill();
-      }
+      setTimeout(() => {
+        if (plugin) {
+          plugin.kill();
+        }
+      }, 5000);
```

Comment:
The thing about reusing the plugin object was that if the user re-enables the plugin before the 5 seconds expires, then we'll wind up killing the new process.

If cleanup did something like this:
```
    const cleanup = () => {
      let process = plugin.process;
      setTimeout(() => {
        process.kill();
      }, 5000);
    };
```
then it wouldn't matter. Otherwise this looks good.